tests/unit/core/test_source_mapping.py

```diff
@@ -102,26 +100,16 @@ def _test_references_user_defined_aliases():
     assert lines == [13, 16]
 
 
-def _test_references_user_defined_types_when_casting():
+def test_references_user_defined_types_when_casting(solc_binary_path):
     """
     Tests if references are filled correctly for user defined types in case of casting.
     """
-    solc_select.switch_global_version("0.8.16", always_install=True)
+    solc_path = solc_binary_path("0.8.16")
     file = Path(SRC_MAPPING_TEST_ROOT, "ReferencesUserDefinedTypesCasting.sol").as_posix()
-    slither = Slither(file)
+    slither = Slither(file, solc=solc_path)
 
     contracts = slither.compilation_units[0].contracts
     a = contracts[0] if contracts[0].is_interface else contracts[1]
     assert len(a.references) == 2
     lines = _sort_references_lines(a.references)
     assert lines == [12, 18]
-
-
-def test_references():
```

Comment:
was this removed?

Reply:
No, the test was calling two nested tests and I split them to be their own test. https://github.com/crytic/slither/pull/1808/files/d1b9e8288900d6b7d75025b3f589adec0b68a045#diff-fee079d1aa52bfd03e4808f11f095beadfb074ecb24718f690004bb271bd4754R80-R103